data-prepper-plugins/aggregate-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/aggregate/AggregateActionSynchronizer.java

```diff
@@ -74,7 +75,7 @@ Optional<Event> concludeGroup(final AggregateIdentificationKeysHasher.Identifica
         return concludeGroupEvent;
     }
 
-    AggregateActionResponse handleEventForGroup(final Event event, final AggregateIdentificationKeysHasher.IdentificationHash hash, final AggregateGroup aggregateGroup) {
+    AggregateActionResponse handleEventForGroup(final Event event, final AggregateIdentificationKeysHasher.IdentificationHash hash, final AggregateGroup aggregateGroup, Map<String, Object> identificationKeysMap) {
```

Comment:
The `AggregateIdentificationKeysHasher.IdentificationHash` already contains the same information as the `identificationKeysMap` you are passing now.

Reply:
Right, But does it make sense to have an API in `AggregateIdentificationKeysHasher`, get the hashMap from it? I wasn't sure. If that's OK, may be, we can pass it to concludeGroup()? That's one option I see.

Reply:
I think it's ok to have a getter to the actual map, or to just create some wrapper functions for that will only allow reading the keys and values